File: 75-sort-colors/sort-colors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function(nums) {
    let low = 0;
    let high = nums.length -1;
    let mid = 0;
    
    while (mid <= high){
        if (nums[mid] == 0){
            [nums[mid], nums[low]] = [nums[low], nums[mid]];
            mid +=1;
            low += 1;
        }
        else if (nums[mid] == 1) mid += 1;
        
        else{
            [nums[mid], nums[high]] = [nums[high], nums[mid]];
            high -= 1;
        }
    }
};

File: 75-sort-colors/sort-colors.dart
class Solution {
  void sortColors(List<int> nums) {
    int low =0;
      int mid = 0;
      int high = nums.length -1;
      
      while (mid <= high){
          
          if (nums[mid] == 0){
              int temp = nums[mid];
              nums[mid] = nums[low];
              nums[low] = temp;
              low +=1 ;
              mid += 1;
          }
          
          else if(nums[mid] == 1) mid +=1;
          
          else{
              int temp = nums[mid];
              nums[mid] = nums[high];
              nums[high] = temp;
              high -=1;
          }
      }
  }
}